src/lib/crypto/kimchi_backend/gadgets/foreign_field.ml

```diff
@@ -0,0 +1,1065 @@
+open Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint
+
+module Bignum_bigint = Snarky_backendless.Backend_extended.Bignum_bigint
+module Snark_intf = Snarky_backendless.Snark_intf
+
+let two_to_limb_field (type f)
+    (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f) =
+  Common.(bignum_bigint_to_field (module Circuit) two_to_limb)
+
+(* 2^2L *)
+let two_to_2limb =
```

Comment:
It's a bignum_bigint rather than a field, but good suggestion.  Thanks.   I'll just square it like so
```ocaml
  let two_to_2limb =
  Bignum_bigint.(pow Common.two_to_limb (of_int 2))
```